src/NetscriptFunctions/Corporation.ts

```diff
@@ -332,10 +332,13 @@ export function NetscriptCorporation(
       const cityName = helper.string("getMaterial", "cityName", acityName);
       const materialName = helper.string("getMaterial", "materialName", amaterialName);
       const material = getMaterial(divisionName, cityName, materialName);
+      const corporation = getCorporation();
       return {
         name: material.name,
         qty: material.qty,
         qlt: material.qlt,
+        dmd: corporation.unlockUpgrades[2] ? material.dmd : undefined,
```

Comment:
is there any constants that define 2 and 3 as the unlocks (just for some more readability)?

Reply:
No. The closest we could get is something like
`corporation.unlockUpgrades[CorporationUnlockUpgrades.find(u=>u[2]==='Market Research - Demand')[0] ]`
Which doesn't seem much better.

A couple places in the code use a "readability" pattern like the following:
`marketResearchDemand = corporation.unlockUpgrades[2];`

Reply:
https://github.com/danielyxie/bitburner/blob/dev/src/NetscriptFunctions/Corporation.ts#L86 hmm i think this is the function

Reply:
It feels a little weird to use a netscript function for this purpose, but I guess it works.

Also, on further consideration... it should also be exposing sllman, so that we can retrieve city-by-city pricing information, and the cityData array isn't being copied, so changes to it made by netscript will affect the quantity and reported production/sales.

Reply:
what is sllman? 😆 i hate shortening variable names

Reply:
?sell manager? I have no idea. It and prdman are used to store the per-city product sell price and production limits. pseudocode:
sllman[cityName] := [isPriceSetInThisCity, priceInThisCity];
prdman[cityName] := [isProductionLimitedInThisCity, productionLimitInThisCity];